OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -267,6 +267,27 @@ public static ProcStatus Enumerate(DMProcState state) {
             return ProcStatus.Continue;
         }
 
+        public static ProcStatus EnumerateAssoc(DMProcState state) {
+            var enumeratorId = state.ReadInt();
+            var outputRef = state.ReadReference();
+            var assocRef = state.ReadReference();
+            var listRef = state.ReadReference();
+            var jumpToIfFailure = state.ReadInt();
+
+            var enumerator = state.Enumerators[enumeratorId];
+            if (enumerator == null || !enumerator.Enumerate(state, outputRef))
+                state.Jump(jumpToIfFailure);
+            else {
+                var outputVal = state.GetReferenceValue(outputRef, peek: true);
+                var listVal = state.GetReferenceValue(listRef, peek: true);
+                var indexVal = state.GetIndex(listVal, outputVal, state);
+
+                state.AssignReference(assocRef, indexVal);
```

Comment:
References can potentially require certain values at the top of the stack, so the order they are gotten matters. They need to be popped in the same order they appear in the opcode arguments, so `outputRef`, `assocRef`, then `listRef`. Using `peek: true` is also leaving the incorrect values on the stack for the next `GetReferenceValue()`.